Updates to caching technique in the GitHub Actions Guide (#3947)
cypress-io/cypress-documentation: content/guides/continuous-integration/github-actions.md

## Caching Dependencies and Build Artifacts

> The Cypress team maintains the official [Cypress GitHub Action](https://github.com/marketplace/actions/cypress-io) for running Cypress tests. This action provides npm installation, custom caching, additional configuration options and simplifies setup of advanced workflows with Cypress in the GitHub Actions platform.

Reviewer: ```suggestion
The Cypress team maintains the official [Cypress GitHub Action](https://github.com/marketplace/actions/cypress-io) for running Cypress tests. This action provides npm installation, custom caching, additional configuration options and simplifies setting up advanced workflows with Cypress in the GitHub Actions platform.
```

<Alert type="info">

> Caching of dependencies and build artifacts between installation and worker jobs can be accomplished with the GitHub Actions [Upload/Download Artifact Actions](https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts).

Reviewer: ```suggestion
Caching of dependencies and build artifacts between installation and worker jobs can be accomplished with the [Upload/Download Artifact Actions](https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts).
```

</Alert>

> The `install` job below uses the [upload-artifact](https://github.com/marketplace/actions/upload-a-build-artifact) GitHub Action is included and will save the state of the `build` directory for the worker jobs.

Reviewer: ```suggestion
The `install` job below uses the [upload-artifact](https://github.com/marketplace/actions/upload-a-build-artifact) action and will save the state of the `build` directory for the worker jobs.
```

```yaml
name: Cypress Tests with installation job

on: [push]

jobs:
  install:
    runs-on: ubuntu-latest
    container: cypress/browsers:node12.18.3-chrome87-ff82
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Save build folder
        uses: actions/upload-artifact@v2
        with:
          name: build
          if-no-files-found: error
          path: build

      - name: Cypress install
        uses: cypress-io/github-action@v2
        with:
          # Disable running of tests within install job
          runTests: false
          build: yarn build
```

> The [download-artifact](https://github.com/marketplace/actions/download-a-build-artifact) GitHub Action is included and will retrieve the `build` directory saved in the install job, as seen below in a worker job.

Reviewer: ```suggestion
The [download-artifact](https://github.com/marketplace/actions/download-a-build-artifact) action will retrieve the `build` directory saved in the install job, as seen below in a worker job.
```

First, we'll define the `install` step that will be used by the worker jobs defined in the matrix strategy.

For the `steps`, notice that we pass `runTests: false` to the Cypress GitHub Action to instruct it to only install and cache Cypress and npm dependencies _without running the tests_.

> The [upload-artifact](https://github.com/marketplace/actions/upload-a-build-artifact) GitHub Action is included and will save the state of the `build` directory for the worker jobs.

Reviewer: ```suggestion
The [upload-artifact](https://github.com/marketplace/actions/upload-a-build-artifact) action will save the state of the `build` directory for the worker jobs.
```

Using our [Cypress GitHub Action](https://github.com/marketplace/actions/cypress-io) we specify `install: false` since our dependencies and build were cached in our `install` job.

> The [download-artifact](https://github.com/marketplace/actions/download-a-build-artifact) GitHub Action is included and will retrieve the `build` directory saved in the install job.

Reviewer: ```suggestion
The [download-artifact](https://github.com/marketplace/actions/download-a-build-artifact) action will retrieve the `build` directory saved in the install job.
```
